NDice/DieLabeller.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NDice
+{
+	/// <summary>
+	/// Labels a Die roll result to a value of type TLabel
+	/// </summary>
+    public class DieLabeller<TLabel>
+    {
+		private readonly IDie die = null;

```

Comment:
No need for assignments. Declaring is fine. Same for the all of the private variables.

Reply:
Also, the convention in this lib is private vars are underscored: `_die`, `_labels`, and `throwsIfLabelMissing`. Then `this` can be removed in the constructors.